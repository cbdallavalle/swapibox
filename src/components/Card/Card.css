h2, span {
  font-family: impact;
}

span {
  color: #ccc8c8;
  font-size: 19px;
}

hr {
  width: 80%;
  border: 1.5px solid white;
}

.cards {
  position: relative;
  background-color: #ffffff3b;
  width: 260px;
  height: fit-content;
  padding-top: 7px;
  border-radius: 19px;
  margin: 10px;
  border: 2px solid white;
}

.cards:hover {
  cursor: pointer;
  background-color: #ffffffad;
  outline: none;
}

.target-img {
  width: 30px;
  position: absolute;
  top: 5px;
  right: 5px;
}

#marked {
  background-color: #9d797752;
  border: 2px solid #A13B37;
}