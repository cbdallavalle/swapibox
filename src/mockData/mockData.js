const mockData = {
  peopleData: {
    name: "Darth Vader",
    info: {
      homeworld: "Tatooine",
      species: "Human",
      homePop: "200000"  
    }

  },
  planetData: {
    name: "Alderaan", 
    info: {
      terrain: "grasslands, mountains", 
      population: "2000000000", 
      climate: "temperate", 
      residents: [
        {resident: "Leia Organa"},
        {resident: "Bail Prestor Organa"},
        {resident: "Raymus Antilles"}
      ]  
    }
  },
  vehicleData: {
    name: "Sand Crawler",
    info: {
      model: "Digger Crawler",
      passengers: "30",
      vehicleClass: "wheeled" 
    }
  },
}

export default mockData;