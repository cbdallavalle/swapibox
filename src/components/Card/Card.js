import React from 'react';
import target from '../../styles/bullseye.svg';
import './Card.css';

const Card = ({card, toggleTarget }) => {
  console.log(card);
  const { name, species, homeworld, homePop, marked } = card;
  const { population, terrain, climate } = card;
  const { model, vehicleClass, passengers } = card

  const cardType = Object.keys(card.info).map( key => {
    return key !== "residents" ? 
      <h4><span>{key}:</span> {card.info[key]}</h4>
      :
      <ul><span>{key}</span> 
      {
        Object.values(card.info.residents).map( value => <li>{value.resident}</li>)
      }
      </ul>
  })

  // const cardType = () => {
  //   if(species) {
  //     return (
  //       <div>
  //         <img className="target-img" src={target} alt="target" />
  //         <h4><span>species:</span> {species}</h4>
  //         <h4><span>homeworld:</span> {homeworld}</h4>
  //         <h4><span>population:</span> {homePop}</h4>
  //       </div>
  //     )
  //   } 
  //   else if (terrain) {
  //     return (
  //       <div>
  //         <img className="target-img" src={target} alt="target" />
  //         <h4><span>population: </span>{population}</h4>
  //         <h4><span>terrain: </span>{terrain}</h4>
  //         <h4><span>climate: </span>{climate}</h4>
  //       </div>
  //     )
  //   } else {
  //     return (
  //       <div>
  //         <h4><span>model:</span> {model}</h4>
  //         <h4><span>class:</span> {vehicleClass}</h4>
  //         <h4><span>passengers:</span> {passengers}</h4>
  //       </div>
  //     )
  //   }
  // }

  return (
    <article className="cards" 
             id={marked}
             onClick={ () => toggleTarget(card) }
    >
      <img className="target-img" src={target} alt="target" />
      <h2>{name}</h2>
      <hr />
      { cardType }
    </article>
  )
}

export default Card;