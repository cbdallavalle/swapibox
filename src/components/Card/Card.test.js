/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';
import mockData from '../../mockData/mockData';

describe('Card', () => {
  let wrapper;
  const mockToggleTarget = jest.fn();

  it('should match snapshot when passed data for people', () => {
    wrapper = shallow(          
      <Card key={ 1 }
            card={ mockData.peopleData }
            toggleTarget={ mockToggleTarget }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when passed data for planets', () => {
    wrapper = shallow(          
      <Card key={ 1 }
            card={ mockData.planetData }
            toggleTarget={ mockToggleTarget }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when passed data for vehicles', () => {
    wrapper = shallow(          
      <Card key={ 1 }
            card={ mockData.vehicleData }
            toggleTarget={ mockToggleTarget }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('when a card is clicked, should call toggleTarget', () => {
    wrapper = shallow(          
      <Card key={ 1 }
            card={ mockData.planetData }
            toggleTarget={ mockToggleTarget }
      />
    )
    wrapper.find('article').simulate('click');
    expect(mockToggleTarget).toHaveBeenCalled();
  })
})