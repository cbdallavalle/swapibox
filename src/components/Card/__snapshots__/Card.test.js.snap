// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot when passed data for people 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "homePop": "200000",
            "homeworld": "Tatooine",
            "name": "Darth Vader",
            "species": "Human",
          }
    }
    toggleTarget={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Darth Vader
</h2>,
        <hr />,
        <div>
          <img
                    alt="target"
                    className="target-img"
                    src="bullseye.svg"
          />
          <h4>
                    <span>
                              species:
                    </span>
                     
                    Human
          </h4>
          <h4>
                    <span>
                              homeworld:
                    </span>
                     
                    Tatooine
          </h4>
          <h4>
                    <span>
                              population:
                    </span>
                     
                    200000
          </h4>
</div>,
      ],
      "className": "cards",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Darth Vader",
        },
        "ref": null,
        "rendered": "Darth Vader",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="target"
              className="target-img"
              src="bullseye.svg"
/>,
            <h4>
              <span>
                            species:
              </span>
               
              Human
</h4>,
            <h4>
              <span>
                            homeworld:
              </span>
               
              Tatooine
</h4>,
            <h4>
              <span>
                            population:
              </span>
               
              200000
</h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "target",
              "className": "target-img",
              "src": "bullseye.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  species:
</span>,
                " ",
                "Human",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "species:",
                },
                "ref": null,
                "rendered": "species:",
                "type": "span",
              },
              " ",
              "Human",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  homeworld:
</span>,
                " ",
                "Tatooine",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "homeworld:",
                },
                "ref": null,
                "rendered": "homeworld:",
                "type": "span",
              },
              " ",
              "Tatooine",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  population:
</span>,
                " ",
                "200000",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "population:",
                },
                "ref": null,
                "rendered": "population:",
                "type": "span",
              },
              " ",
              "200000",
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Darth Vader
</h2>,
          <hr />,
          <div>
            <img
                        alt="target"
                        className="target-img"
                        src="bullseye.svg"
            />
            <h4>
                        <span>
                                    species:
                        </span>
                         
                        Human
            </h4>
            <h4>
                        <span>
                                    homeworld:
                        </span>
                         
                        Tatooine
            </h4>
            <h4>
                        <span>
                                    population:
                        </span>
                         
                        200000
            </h4>
</div>,
        ],
        "className": "cards",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Darth Vader",
          },
          "ref": null,
          "rendered": "Darth Vader",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="target"
                className="target-img"
                src="bullseye.svg"
/>,
              <h4>
                <span>
                                species:
                </span>
                 
                Human
</h4>,
              <h4>
                <span>
                                homeworld:
                </span>
                 
                Tatooine
</h4>,
              <h4>
                <span>
                                population:
                </span>
                 
                200000
</h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "target",
                "className": "target-img",
                "src": "bullseye.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    species:
</span>,
                  " ",
                  "Human",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "species:",
                  },
                  "ref": null,
                  "rendered": "species:",
                  "type": "span",
                },
                " ",
                "Human",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    homeworld:
</span>,
                  " ",
                  "Tatooine",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "homeworld:",
                  },
                  "ref": null,
                  "rendered": "homeworld:",
                  "type": "span",
                },
                " ",
                "Tatooine",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    population:
</span>,
                  " ",
                  "200000",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "population:",
                  },
                  "ref": null,
                  "rendered": "population:",
                  "type": "span",
                },
                " ",
                "200000",
              ],
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot when passed data for planets 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "climate": "temperate",
            "name": "Alderaan",
            "population": "2000000000",
            "residents": Array [
              Object {
                "resident": "Leia Organa",
              },
              Object {
                "resident": "Bail Prestor Organa",
              },
              Object {
                "resident": "Raymus Antilles",
              },
            ],
            "terrain": "grasslands, mountains",
          }
    }
    toggleTarget={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Alderaan
</h2>,
        <hr />,
        <div>
          <img
                    alt="target"
                    className="target-img"
                    src="bullseye.svg"
          />
          <h4>
                    <span>
                              population: 
                    </span>
                    2000000000
          </h4>
          <h4>
                    <span>
                              terrain: 
                    </span>
                    grasslands, mountains
          </h4>
          <h4>
                    <span>
                              climate: 
                    </span>
                    temperate
          </h4>
</div>,
      ],
      "className": "cards",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alderaan",
        },
        "ref": null,
        "rendered": "Alderaan",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="target"
              className="target-img"
              src="bullseye.svg"
/>,
            <h4>
              <span>
                            population: 
              </span>
              2000000000
</h4>,
            <h4>
              <span>
                            terrain: 
              </span>
              grasslands, mountains
</h4>,
            <h4>
              <span>
                            climate: 
              </span>
              temperate
</h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "target",
              "className": "target-img",
              "src": "bullseye.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  population: 
</span>,
                "2000000000",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "population: ",
                },
                "ref": null,
                "rendered": "population: ",
                "type": "span",
              },
              "2000000000",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  terrain: 
</span>,
                "grasslands, mountains",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "terrain: ",
                },
                "ref": null,
                "rendered": "terrain: ",
                "type": "span",
              },
              "grasslands, mountains",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  climate: 
</span>,
                "temperate",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "climate: ",
                },
                "ref": null,
                "rendered": "climate: ",
                "type": "span",
              },
              "temperate",
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Alderaan
</h2>,
          <hr />,
          <div>
            <img
                        alt="target"
                        className="target-img"
                        src="bullseye.svg"
            />
            <h4>
                        <span>
                                    population: 
                        </span>
                        2000000000
            </h4>
            <h4>
                        <span>
                                    terrain: 
                        </span>
                        grasslands, mountains
            </h4>
            <h4>
                        <span>
                                    climate: 
                        </span>
                        temperate
            </h4>
</div>,
        ],
        "className": "cards",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alderaan",
          },
          "ref": null,
          "rendered": "Alderaan",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="target"
                className="target-img"
                src="bullseye.svg"
/>,
              <h4>
                <span>
                                population: 
                </span>
                2000000000
</h4>,
              <h4>
                <span>
                                terrain: 
                </span>
                grasslands, mountains
</h4>,
              <h4>
                <span>
                                climate: 
                </span>
                temperate
</h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "target",
                "className": "target-img",
                "src": "bullseye.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    population: 
</span>,
                  "2000000000",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "population: ",
                  },
                  "ref": null,
                  "rendered": "population: ",
                  "type": "span",
                },
                "2000000000",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    terrain: 
</span>,
                  "grasslands, mountains",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "terrain: ",
                  },
                  "ref": null,
                  "rendered": "terrain: ",
                  "type": "span",
                },
                "grasslands, mountains",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    climate: 
</span>,
                  "temperate",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "climate: ",
                  },
                  "ref": null,
                  "rendered": "climate: ",
                  "type": "span",
                },
                "temperate",
              ],
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot when passed data for vehicles 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
            "vehicleClass": "wheeled",
          }
    }
    toggleTarget={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sand Crawler
</h2>,
        <hr />,
        <div>
          <h4>
                    <span>
                              model:
                    </span>
                     
                    Digger Crawler
          </h4>
          <h4>
                    <span>
                              class:
                    </span>
                     
                    wheeled
          </h4>
          <h4>
                    <span>
                              passengers:
                    </span>
                     
                    30
          </h4>
</div>,
      ],
      "className": "cards",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sand Crawler",
        },
        "ref": null,
        "rendered": "Sand Crawler",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              <span>
                            model:
              </span>
               
              Digger Crawler
</h4>,
            <h4>
              <span>
                            class:
              </span>
               
              wheeled
</h4>,
            <h4>
              <span>
                            passengers:
              </span>
               
              30
</h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  model:
</span>,
                " ",
                "Digger Crawler",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "model:",
                },
                "ref": null,
                "rendered": "model:",
                "type": "span",
              },
              " ",
              "Digger Crawler",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  class:
</span>,
                " ",
                "wheeled",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "class:",
                },
                "ref": null,
                "rendered": "class:",
                "type": "span",
              },
              " ",
              "wheeled",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  passengers:
</span>,
                " ",
                "30",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "passengers:",
                },
                "ref": null,
                "rendered": "passengers:",
                "type": "span",
              },
              " ",
              "30",
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sand Crawler
</h2>,
          <hr />,
          <div>
            <h4>
                        <span>
                                    model:
                        </span>
                         
                        Digger Crawler
            </h4>
            <h4>
                        <span>
                                    class:
                        </span>
                         
                        wheeled
            </h4>
            <h4>
                        <span>
                                    passengers:
                        </span>
                         
                        30
            </h4>
</div>,
        ],
        "className": "cards",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sand Crawler",
          },
          "ref": null,
          "rendered": "Sand Crawler",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                <span>
                                model:
                </span>
                 
                Digger Crawler
</h4>,
              <h4>
                <span>
                                class:
                </span>
                 
                wheeled
</h4>,
              <h4>
                <span>
                                passengers:
                </span>
                 
                30
</h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    model:
</span>,
                  " ",
                  "Digger Crawler",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "model:",
                  },
                  "ref": null,
                  "rendered": "model:",
                  "type": "span",
                },
                " ",
                "Digger Crawler",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    class:
</span>,
                  " ",
                  "wheeled",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "class:",
                  },
                  "ref": null,
                  "rendered": "class:",
                  "type": "span",
                },
                " ",
                "wheeled",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    passengers:
</span>,
                  " ",
                  "30",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "passengers:",
                  },
                  "ref": null,
                  "rendered": "passengers:",
                  "type": "span",
                },
                " ",
                "30",
              ],
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
